pkg/upbin/upbin.go

```diff
@@ -0,0 +1,398 @@
+	installPackageManagerFunc = installPackageManager
+)
+
+func UpdateBin() error {
+	notfy, err := notifyFunc()
+	if err != nil {
+		return err
+	}
+
+	if !notfy {
+		return nil
+	}
+
+	if !needToUpdateFunc() {
+		return nil
+	}
+
+	if !wantToUpdateFunc() {
+		return saveLastActivityFunc()
+	}
+
+	fileURL, err := prepareURLFunc()
+	if err != nil {
+		return err
+	}
+
+	install, err := managersPackagesFunc()
+	if err != nil {
+		return err
+	}
+
+	if install {
+		return nil
+	}
+
+	filePath, _ := whichFunc(azioncli)
+
+	err = downloadFileFunc(filePath, fileURL)
+	if err != nil {
+		return err
+	}
+
+	err = replaceFileFunc(filePath)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func GetCurrentVersion() int {
+	n, _ := Format(version.BinVersion)
+	return n
+}
+
+func GetLatestVersion() int {
+	r, _ := git.Clone(memory.NewStorage(), nil, &git.CloneOptions{URL: repoURL})
+
+	tags, _ := r.Tags()
+	t, _ := latestTag(tags)
+
+	n, _ := Format(t)
+	return n
+}
+
+func Format(input string) (int, error) {
+	numberString := ""
+	for _, char := range input {
+		if unicode.IsDigit(char) {
+			numberString += string(char)
+		}
+	}
+
+	number, err := strconv.Atoi(numberString)
+	if err != nil {
+		return 0, err
+	}
+
+	return number, nil
+}
+
+type ReferenceIter interface {
+	ForEach(func(*plumbing.Reference) error) error
+}
+
+// latestTag return value in format refs/tags/v0.10.0
+func latestTag(tags ReferenceIter) (tag string, err error) {
+	var biggerVersionSoFar int = 0
+
+	err = tags.ForEach(func(t *plumbing.Reference) error {
+		tagCurrent := t.Name().String() // return this format "refs/tags/v0.10.0"
+
+		if !strings.Contains(tagCurrent, "dev") {
+			versionParts := strings.Split(tagCurrent, ".")
+
+			major := strings.TrimPrefix(versionParts[0], "refs/tags/v")
+			minor := versionParts[1]
+			patch := versionParts[2]
+
+			current, _ := strconv.Atoi(fmt.Sprintf("%s%s%s", major, minor, patch))
+
+			if current > biggerVersionSoFar {
+				biggerVersionSoFar = current
+				tag = tagCurrent
+			}
+		}
+
+		return err
+	})
+
+	return tag, err
+}
+
+func which(command string) (string, error) {
+	path := os.Getenv("PATH")
+	paths := filepath.SplitList(path)
+
+	for _, dir := range paths {
+		executablePath := filepath.Join(dir, command)
+		_, err := os.Stat(executablePath)
+		if err == nil {
+			return executablePath, nil
+		}
+	}
+
+	return "", fmt.Errorf("command '%s' not found", command)
+}
+
+func GetSystem() system {
+	switch system(runtime.GOOS) {
+	case linux:
+		return linux
+	case darwin:
+		return darwin
+	case windows:
+		return windows
+	default:
+		return unknown
+	}
+}
+
+func prepareURL() (string, error) {
+	sys := GetSystem()
+	if sys == unknown {
+		return "", errors.New("unknown system")
+	}
+	return fmt.Sprintf(url, sys), nil
+}
+
+func needToUpdate() bool {
+	return GetLatestVersion() >= GetCurrentVersion()
+}
+
+func wantToUpdate() bool {
+	prompt := promptui.Select{
+		Label: "Do you want to update 'azioncli'?",
```

Comment:
Maybe:
"A new version of 'azioncli' was published. Do you wish to update to the latest version?"
or something like that